core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -1193,6 +1193,18 @@ public void getState_loadBalancerDoesNotSupportChannelState() {
   }
 
   @Test
+  public void getState_loadBalancerDoesNotSupportChannelStateThenChannelShutdown() {
+    createChannel(new FakeNameResolverFactory(false), NO_INTERCEPTOR);
+    assertEquals(ConnectivityState.IDLE, channel.getState(false));
+    helper.updatePicker(mockPicker);
+
+    channel.shutdown();
```

Comment:
Applications may also turn on `-ea`, then the entire application may fail if any runtime exception is thrown in `ChannelExecutor`'s runnables.